eclair-core/src/main/scala/fr/acinq/eclair/channel/publish/ReplaceableTxPrePublisher.scala

```diff
@@ -0,0 +1,283 @@
+    override def updateTx(tx: Transaction): ClaimHtlcWithWitnessData
+  }
+  case class ClaimHtlcSuccessWithWitnessData(txInfo: ClaimHtlcSuccessTx, preimage: ByteVector32) extends ClaimHtlcWithWitnessData {
+    override def updateTx(tx: Transaction): ClaimHtlcSuccessWithWitnessData = this.copy(txInfo = this.txInfo.copy(tx = tx))
+  }
+  case class LegacyClaimHtlcSuccessWithWitnessData(txInfo: LegacyClaimHtlcSuccessTx, preimage: ByteVector32) extends ClaimHtlcWithWitnessData {
+    override def updateTx(tx: Transaction): LegacyClaimHtlcSuccessWithWitnessData = this.copy(txInfo = this.txInfo.copy(tx = tx))
+  }
+  case class ClaimHtlcTimeoutWithWitnessData(txInfo: ClaimHtlcTimeoutTx) extends ClaimHtlcWithWitnessData {
+    override def updateTx(tx: Transaction): ClaimHtlcTimeoutWithWitnessData = this.copy(txInfo = this.txInfo.copy(tx = tx))
+  }
+  // @formatter:on
+
+  def apply(nodeParams: NodeParams, bitcoinClient: BitcoinCoreClient, loggingInfo: TxPublishLogContext): Behavior[Command] = {
+    Behaviors.setup { context =>
+      Behaviors.withMdc(loggingInfo.mdc()) {
+        Behaviors.receiveMessagePartial {
+          case CheckPreconditions(replyTo, cmd) =>
+            val prePublisher = new ReplaceableTxPrePublisher(nodeParams, replyTo, cmd, bitcoinClient, context)
+            cmd.txInfo match {
+              case localAnchorTx: Transactions.ClaimLocalAnchorOutputTx => prePublisher.checkAnchorPreconditions(localAnchorTx)
+              case htlcTx: Transactions.HtlcTx => prePublisher.checkHtlcPreconditions(htlcTx)
+              case claimHtlcTx: Transactions.ClaimHtlcTx => prePublisher.checkClaimHtlcPreconditions(claimHtlcTx)
+            }
+        }
+      }
+    }
+  }
+
+}
+
+private class ReplaceableTxPrePublisher(nodeParams: NodeParams,
+                                        replyTo: ActorRef[ReplaceableTxPrePublisher.PreconditionsResult],
+                                        cmd: TxPublisher.PublishReplaceableTx,
+                                        bitcoinClient: BitcoinCoreClient,
+                                        context: ActorContext[ReplaceableTxPrePublisher.Command])(implicit ec: ExecutionContext = ExecutionContext.Implicits.global) {
+
+  import ReplaceableTxPrePublisher._
+
+  private val log = context.log
+
+  def checkAnchorPreconditions(localAnchorTx: ClaimLocalAnchorOutputTx): Behavior[Command] = {
+    // We verify that:
+    //  - our commit is not confirmed (if it is, no need to claim our anchor)
+    //  - their commit is not confirmed (if it is, no need to claim our anchor either)
+    //  - our commit tx is in the mempool (otherwise we can't claim our anchor)
+    val commitTx = cmd.commitments.fullySignedLocalCommitTx(nodeParams.channelKeyManager).tx
+    val fundingOutpoint = cmd.commitments.commitInput.outPoint
+    context.pipeToSelf(bitcoinClient.isTransactionOutputSpendable(fundingOutpoint.txid, fundingOutpoint.index.toInt, includeMempool = false).flatMap {
```

Comment:
I would use `getTxConfirmations` instead. The `isTransactionOutputSpendable` is footgunny because it returns `false` when the utxo doesn't exist at all. In a scenario where we force close in `WAIT_FOR_FUNDING_CONFIRMED` we would think the commitment tx has been confirmed when it has actually been double spent.

Reply:
Done in https://github.com/ACINQ/eclair/pull/2113/commits/e389af61d928d9fa921ba90d400709b1d8ea4543